Code/Framework/AzToolsFramework/AzToolsFramework/Input/QtEventToAzInputMapper.cpp

```diff
@@ -408,32 +439,76 @@ namespace AzToolsFramework
+                // down events.  dispatching the key event as text on shortcut (and auto-repeat press) ensures
+                // all printable keys a fair chance at being consumed before processing elsewhere
+                if (eventType == QEvent::Type::ShortcutOverride ||
+                    (eventType == QEvent::Type::KeyPress && keyEvent->isAutoRepeat()))
+                {
+                    bool textConsumed = false;
+                    AzFramework::InputTextNotificationBus::Broadcast(
+                        &AzFramework::InputTextNotifications::OnInputTextEvent,
+                        AZStd::string(keyText.toUtf8().data()),
+                        textConsumed
+                    );
+                    consumeAsText = textConsumed;
+                }
+                // follow up events from accepting the previous key event also need to be accepted without
+                // dispatching the text event, otherwise the key event will be processed twice: once for the
+                // previous text event, then again as a normal key event which can lead to odd behaviour
+                // during text edit
+                else
+                {
+                    consumeAsText &= (lastKey == key);
+                }
+            }
+            else
+            {
+                consumeAsText = false;
+            }
+
+            lastKey = key;
+
+            if (consumeAsText)
+            {
+                keyEvent->accept();
+                return;
+            }
+        }
+
+        // Ignore key repeat events for non-text, they're unrelated to actual physical button presses.
         if (keyEvent->isAutoRepeat())
         {
             return;
         }
 
-        const Qt::Key key = static_cast<Qt::Key>(keyEvent->key());
-
         // For ShortcutEvent, only continue processing if we're in the HighPriorityKeys set.
-        if (keyEvent->type() != QEvent::Type::ShortcutOverride || m_highPriorityKeys.find(key) != m_highPriorityKeys.end())
+        if (eventType != QEvent::Type::ShortcutOverride || m_highPriorityKeys.find(key) != m_highPriorityKeys.end())
         {
             if (auto keyIt = m_keyMappings.find(key); keyIt != m_keyMappings.end())
             {
                 auto keyChannel = GetInputChannel<AzFramework::InputChannelDigitalWithSharedModifierKeyStates>(keyIt->second);
 
                 if (keyChannel)
                 {
-                    if (keyEvent->type() == QEvent::Type::KeyPress || keyEvent->type() == QEvent::Type::ShortcutOverride)
-                    {
-                        keyChannel->UpdateState(true);
-                    }
-                    else
-                    {
-                        keyChannel->UpdateState(false);
-                    }
-
+                    const bool isKeyPressed = (eventType == QEvent::Type::KeyPress) || (eventType == QEvent::Type::ShortcutOverride);
```

Comment:
nit: redundant parens (this is sort of style thing so you can ignore but they weren't there before this change - sometimes they are of course needed, but otherwise they just add additional visual noise and clutter and don't add anything).